cmd/dbos/workflow.go

```diff
@@ -0,0 +1,411 @@
+		case "ERROR":
+			statusType = dbos.WorkflowStatusError
+		case "ENQUEUED":
+			statusType = dbos.WorkflowStatusEnqueued
+		case "CANCELLED":
+			statusType = dbos.WorkflowStatusCancelled
+		case "MAX_RECOVERY_ATTEMPTS_EXCEEDED":
+			statusType = dbos.WorkflowStatusMaxRecoveryAttemptsExceeded
+		default:
+			return fmt.Errorf("invalid status: %s", status)
+		}
+		opts = append(opts, dbos.WithStatus([]dbos.WorkflowStatusType{statusType}))
+	}
+
+	if appVersion, _ := cmd.Flags().GetString("application-version"); appVersion != "" {
+		opts = append(opts, dbos.WithAppVersion(appVersion))
+	}
+
+	if queue, _ := cmd.Flags().GetString("queue"); queue != "" {
+		opts = append(opts, dbos.WithQueueName(queue))
+	}
+
+	if queuesOnly, _ := cmd.Flags().GetBool("queues-only"); queuesOnly {
+		opts = append(opts, dbos.WithQueuesOnly())
+	}
+
+	if sortDesc, _ := cmd.Flags().GetBool("sort-desc"); sortDesc {
+		opts = append(opts, dbos.WithSortDesc())
+	}
+
+	if startTime, _ := cmd.Flags().GetString("start-time"); startTime != "" {
+		t, err := time.Parse(time.RFC3339, startTime)
+		if err != nil {
+			return fmt.Errorf("invalid start-time format: %w", err)
+		}
+		opts = append(opts, dbos.WithStartTime(t))
+	}
+
+	if endTime, _ := cmd.Flags().GetString("end-time"); endTime != "" {
+		t, err := time.Parse(time.RFC3339, endTime)
+		if err != nil {
+			return fmt.Errorf("invalid end-time format: %w", err)
+		}
+		opts = append(opts, dbos.WithEndTime(t))
+	}
+
+	// List workflows
+	workflows, err := ctx.ListWorkflows(ctx, opts...)
+	if err != nil {
+		return fmt.Errorf("failed to list workflows: %w", err)
+	}
+
+	// Ensure we have a non-nil slice for JSON output
+	if workflows == nil {
+		workflows = []dbos.WorkflowStatus{}
+	}
+
+	// Output results
+	if jsonOutput {
+		return outputJSON(workflows)
+	}
+
+	// Pretty print for non-JSON output
+	if len(workflows) == 0 {
+		logger.Info("No workflows found")
+		return nil
+	}
+
+	for _, wf := range workflows {
+		fmt.Printf("ID: %s\n", wf.ID)
+		fmt.Printf("  Name: %s\n", wf.Name)
+		fmt.Printf("  Status: %s\n", wf.Status)
+		fmt.Printf("  Created: %s\n", wf.CreatedAt.Format(time.RFC3339))
+		if wf.QueueName != "" {
+			fmt.Printf("  Queue: %s\n", wf.QueueName)
+		}
+		fmt.Println()
+	}
+
+	return nil
+}
+
+func runWorkflowGet(cmd *cobra.Command, args []string) error {
+	workflowID := args[0]
+
+	// Get database URL
+	dbURL, err := getDBURL(cmd)
+	if err != nil {
+		return err
+	}
+
+	// Create DBOS context
+	ctx, err := createDBOSContext(dbURL)
+	if err != nil {
+		return err
+	}
+
+	// Retrieve workflow
+	handle, err := ctx.RetrieveWorkflow(ctx, workflowID)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve workflow: %w", err)
+	}
+
+	// Get status
+	status, err := handle.GetStatus()
+	if err != nil {
+		return fmt.Errorf("failed to get workflow status: %w", err)
+	}
+
+	// Output results
+	if jsonOutput {
+		return outputJSON(status)
+	}
+
+	// Pretty print
+	fmt.Printf("Workflow ID: %s\n", status.ID)
```

Comment:
In the interest of maintainability (having learned this lesson in the other CLIs) I would get rid of all the non-JSON output formatting and only emit JSON.